docs/indexing_documents.rst

```diff
@@ -0,0 +1,83 @@
+Indexing Documents
+==================
+
+Elasticsearch accepts documents in the form of JSON. In `elasticsearch-lua`_
+documents are passed as Lua tables. There are several ways to index documents.
+
+Indexing Single Document
+------------------------
+
+To index a document, you need to pass **index** and **type** as parameters.
+The document to be indexed is passed as a Lua table using the **body**
+parameter. Optionally, you can also provide an **id** for the document or let
+Elasticsearch generate it for you.
+
+.. code-block:: lua
+
+  local res, status = client:index{
+    index = "my_index",
+    type = "my_type",
+    id = "my_id",            -- Optional
+    body = {
+      my_key = "my_value"
+    }
+  }
+
+You can specify additional parameters such as **routing**, **refresh**, etc.
```

Comment:
Yes I know, but by seeing this here, people get interested in exploring this feature. I would prefer they explore other parameters, such as `write_consistency` instead of routing.